asllib/ASTUtils.ml

```diff
@@ -804,3 +804,33 @@ let rename_locals map_name ast =
     map_desc_st' d @@ function D_Func f -> D_Func (map_func f) | d -> d
   in
   List.map map_decl ast
+
+let desugar_setter call fields rhs =
+  let loc = to_pos call and { desc = name, args } = call in
+  let () = assert (loc.version = V1) in
+  let here desc = add_pos_from loc desc in
+  match fields with
+  | [] -> S_Call (name, rhs :: args, []) (* Setter(rhs, ...); *)
+  | _ ->
+      let temp = fresh_var "__setter_v1_temporary" in
+      (* temp = Getter(...); *)
+      let read =
+        let getter_call = E_Call (name, args, []) |> here in
+        S_Decl (LDK_Var, LDI_Var temp, Some getter_call) |> here
```

Comment:
Maybe use `LDK_Let` instead of `LDK_Var`?

Reply:
My understanding was that because we then then modify `temp`, it needs to be `LDK_Var`?

Reply:
Oh, sorry. I thought it wasn't being modified.